custom_components/kia_uvo/coordinator.py

```diff
@@ -159,63 +160,83 @@ async def async_check_and_refresh_token(self):
             self.vehicle_manager.check_and_refresh_token
         )
 
+    async def async_wait_for_action_completed(
+        self, vehicle_id: int, action_id: str, interval: int = 5, max_count: int = 6
+    ):
+        for count in range(1, max_count + 1):
+            _LOGGER.debug(f"Last action check: waiting {interval} seconds")
+            await asyncio.sleep(interval)
+            _LOGGER.debug(f"Last action check: attempt {count} of {max_count}")
+            is_completed = await self.hass.async_add_executor_job(
+                self.vehicle_manager.check_action_status, vehicle_id, action_id
+            )
+            _LOGGER.debug(f"Last action completed: {is_completed}")
+            if is_completed:
+                await self.async_refresh()
+                break
+        return is_completed
+
     async def async_lock_vehicle(self, vehicle_id: str):
         await self.async_check_and_refresh_token()
-        await self.hass.async_add_executor_job(self.vehicle_manager.lock, vehicle_id)
-        await self.async_request_refresh()
+        action_id = await self.hass.async_add_executor_job(
```

Comment:
With action ID stored in the coordinator this won't work for multiple cars as it will be overwritten.  So maybe we store the action ID within vehicle in the API and purely call check last action without the ID?

Reply:
Good catch! I realized we may have the same issue with last_pauth so now storing them both per vehicle